### DISCUSSION TITLE
How to run a generator in a workspace context

### DISCUSSION THREAD

**Discussion by moltar** [vercel/turborepo#7541] [Help] [answered]
### Summary

Is it possible to run the generator with the base being inside a workspace, not in the repo root?

E.g. let's say I want to create a React component inside a specific package, how would I go about it?

**Comment by anthonyshew** [comment, marked as answer]
You can! Documentation here: https://turbo.build/repo/docs/core-concepts/monorepos/code-generation#getting-started

**Reply by moltar** [reply]
Sorry, wasn't fully clear on what I meant.

If I define the generators in the root of the repo, how can I run them against a specific workspace repo?

When I cd into a workspace, it doesn't display options from the root.

Does that mean I need to duplicate generators into every workspace?

**Reply by moltar** [reply]
Hm, actually, never mind, I tried this with one of the example repos (`npx create-turbo -e with-changesets`) and when I cd into a package, `turbo gen` does pick up generators from the root.

I must have some weird setup on my project that it does not happen. Will debug.

Thanks for confirming it works!

**Reply by anthonyshew** [reply]
Apologies if I misread!

Interesting that it's not working as intended in your repository. If you think you've found a bug, do please report it into an issue for us so we can get it fixed!
